packages/shared/src/components/MainLayout.tsx

```diff
@@ -158,7 +158,13 @@ export default function MainLayout({
   useEffect(() => {
     if (!shouldRedirectOnboarding) return;
 
-    router.push(`${webappUrl}/onboarding`);
+    const url = new URLSearchParams();
+
+    Object.entries(router.query).forEach(([key, value]) => {
+      url.append(key, value as string);
+    });
+
+    router.push(`${webappUrl}/onboarding?${url.toString()}`);
```

Comment:
I think more efficient would be to use Object argument for router.push, it will avoid forEach and manual concatenation of url. Currently the `?` would be always hanging with or without query params.

```js
router.push({
  pathname: `${webappUrl}/onboarding`,
  query: router.query,
});
```

Reply:
Initially did this but it wasn't working with the extension.

Reply:
Yes, extension is not really a nextjs app which makes it harder. Good catch @sshanzel!

Reply:
Yeah, but you are using `router.push` still, it would not work even for that right?

Reply:
If it is a simple redirection (passing just a string param), it seems to work with `router.push`.

Reply:
Weird are we polyfilling `router` in extension maybe and then it supports only string, because `useRouter` should not even work since there is not nextjs in extension context right?

Reply:
No, we do not polyfill but it works in simple cases such as redirecting.

Reply:
Would not work for the webapp in that case. URL constructor requires a valid http/s URL. And the `webappUrl` is just `/` in the webapp environment. I'd say we move forward with this.

Reply:
Oh man, didn't now that, seem clunky to call it URL then. Yeah, agreed let's merge, sorry for drilling it just looked as something we can make better. 🚀

Reply:
No, it is cool. Would have chosen the URL path too if only the webapp url works on both cases. It ensures everything would go as intended. I trust native classes than my own code 😂